microsoft/testsuites/cpu/common.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from __future__ import annotations
+
+from typing import Dict, List
+
+from lisa import BadEnvironmentStateException, Node
+from lisa.tools import Cat, Dmesg, Echo, Lscpu, Lsvmbus, Uname
+from lisa.util import SkippedException
+
+
+class CPUState:
+    OFFLINE: str = "0"
+    ONLINE: str = "1"
+
+
+def skip_test(node: Node) -> None:
+    uname = node.tools[Uname]
+    kernel_version = uname.get_linux_information().kernel_version
+    config_path = f"/boot/config-{kernel_version}"
+    config_result = node.execute(f"grep CONFIG_HOTPLUG_CPU=y {config_path}", shell=True)
+    if config_result.exit_code != 0:
+        raise SkippedException("This distro doesn't support cpu hot plug.")
+
+
+def set_vmbus_channels_target_cpu(node: Node) -> Dict[str, str]:
+    uname = node.tools[Uname]
+    kernel_version = uname.get_linux_information().kernel_version
+    dmesg = node.tools[Dmesg]
+    lsvmbus = node.tools[Lsvmbus]
+    vmbus_version = dmesg.get_vmbus_version()
+    file_path_list: Dict[str, str] = {}
+    if kernel_version >= "5.8.0" and vmbus_version >= "4.1.0":
+        # save the raw cpu number for each channel for restoring later.
+        channels = lsvmbus.get_device_channels_from_lsvmbus(force_run=True)
+        for channel in channels:
+            for channel_vp_map in channel.channel_vp_map:
+                target_cpu = channel_vp_map.target_cpu
+                if target_cpu == "0":
+                    continue
+                file_path_list[
+                    get_interrupts_assigned_to_cpu(
+                        channel.device_id, channel_vp_map.rel_id
+                    )
+                ] = target_cpu
+        # set all vmbus channel interrupts go into cpu 0.
+        set_cpu_interrupts_assigned_to(file_path_list, node, "0")
+        return file_path_list
+
+
+def get_idle_cpu_set(node: Node) -> List[str]:
+    lsvmbus = node.tools[Lsvmbus]
+    channels = lsvmbus.get_device_channels_from_lsvmbus(force_run=True)
```

Comment:
the method `get_device_channels_from_lsvmbus` is in lsvmbus, it doesn't need to say `from_lsvmbus`.

Reply:
updated